pkg/build/reaper/reaper_test.go

```diff
@@ -0,0 +1,174 @@
+		return &(kerrors.NewNotFound("BuildConfig", "config").(*kerrors.StatusError).ErrStatus)
+	}
+
+	tests := map[string]struct {
+		name     string
+		oc       *testclient.Fake
+		expected []ktestclient.Action
+		err      bool
+	}{
+		"simple stop": {
+			name: "config",
+			oc:   newBuildListFake(makeBuildConfig(1)),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+			},
+			err: false,
+		},
+		"multiple builds": {
+			name: "config",
+			oc:   newBuildListFake(makeBuildConfig(4), mkbuildlist(4)),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewDeleteAction("builds", "default", "build-1"),
+				ktestclient.NewDeleteAction("builds", "default", "build-3"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+				ktestclient.NewDeleteAction("builds", "default", "build-2"),
+				ktestclient.NewDeleteAction("builds", "default", "build-4"),
+			},
+			err: false,
+		},
+		"no config, some builds": {
+			name: "config",
+			oc:   newBuildListFake(mkbuildlist(2)),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewDeleteAction("builds", "default", "build-1"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+				ktestclient.NewDeleteAction("builds", "default", "build-2"),
+			},
+			err: false,
+		},
+		"no config, no builds": {
+			name: "config",
+			oc:   testclient.NewSimpleFake(notFound()),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+			},
+			err: true,
+		},
+		"config, no builds": {
+			name: "config",
+			oc:   testclient.NewSimpleFake(makeBuildConfig(5)),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+			},
+			err: false,
+		},
+	}
+
+	for testName, test := range tests {
+		reaper := &BuildConfigReaper{oc: test.oc, pollInterval: time.Millisecond, timeout: time.Millisecond}
+		_, err := reaper.Stop("default", test.name, 1*time.Second, nil)
```

Comment:
Hmm you bought the idea to hard code the namespace, "default", but now we have all tests with the same `name: "config"`...
Can we tell that all tests should likely run under the same namespace and name? IOW, does it matter for what is being tested here to be able to use a different namespace/name?